source/RialtoGStreamerMSEVideoSink.cpp

```diff
@@ -75,6 +75,33 @@ static GstStateChangeReturn rialto_mse_video_sink_change_state(GstElement *eleme
     return result;
 }
 
+static firebolt::rialto::IMediaPipeline::MediaSource create_media_source(GstCaps *caps)
+{
+    GstStructure *structure = gst_caps_get_structure(caps, 0);
+    const gchar *strct_name = gst_structure_get_name(structure);
+    if (strct_name)
+    {
+        if (g_str_has_prefix(strct_name, "video/x-h264"))
+        {
+            firebolt::rialto::IMediaPipeline::MediaSource viddat(-1, firebolt::rialto::MediaSourceType::VIDEO,
+                                                                 "video/x-h264");
+            viddat.setSegmentAlignment(firebolt::rialto::SegmentAlignment::NAL);
+            return viddat;
+        }
+        else if (g_str_has_prefix(strct_name, "video/mpeg"))
+        {
+            return firebolt::rialto::IMediaPipeline::MediaSource(-1, firebolt::rialto::MediaSourceType::VIDEO,
+                                                                 "video/mpeg2");
+        }
+
```

Comment:
For debugging would it be helpful to have an else clause on here?

else
{
    log_error("Unknown MIME type");
}